tests/integration-tests/tests/disable_hyperthreading/test_disable_hyperthreading.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import pytest
+
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from tests.common.assertions import assert_no_errors_in_logs
+from tests.common.schedulers_common import get_scheduler_commands
+from tests.common.utils import fetch_instance_slots
+
+
+@pytest.mark.regions(["us-east-1"])
+def test_disable_hyperthreading(region, scheduler, instance, os, pcluster_config_reader, clusters_factory):
+    """Test Disable Hyperthreading"""
+    slots_per_instance = fetch_instance_slots(region, instance)
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    _test_disable_hyperthreading(remote_command_executor, scheduler_commands, slots_per_instance)
+
+    assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
+
+
+def _test_disable_hyperthreading(remote_command_executor, scheduler_commands, slots_per_instance):
+    # Test disable hyperthreading on Master
+    logging.info("Test Disable Hyperthreading on Master")
+    result = remote_command_executor.run_remote_command("lscpu")
+    assert_that(result.stdout).contains("Thread(s) per core:    1")
+    assert_that(result.stdout).contains("CPU(s):                {0}".format(slots_per_instance // 2))
```

Comment:
Minor: `t2`s only have 1 thread per core. Probably should make a note that this test should not be run with `t2` instances.

Reply:
I've added a comment and also pytest marker `@pytest.mark.skip_instances(["t2.micro"])`